### PR TITLE
Initial AVS 0.11.0 support

### PR DESCRIPTION
**Pull request by dwelch-spike**
This PR makes asvec compatible with AVS 0.11.0's grpc protocol by pulling in its protobuf spec and making relevant changes. It also adds a few features, they are...
- The ability to set enableVectorIntegrityCheck through index create and index update aswell as view it in index ls vec-395
- Adds --hnsw-batch-reindex-interval and --hnsw-batch-max-reindex-records to index create and index update, also exposes them in index ls. vec-402
- BREAKING: Renames --hnsw-batch-max-records to --hnsw-batch-max-index-records and --hnsw-batch-interval to --hnsw-batch-index-interval to reflect renames in the protobuf. vec-403
- Exposes indexed vector records and vertices in index ls. vec-396

Still to come in another PR
- Support for record caching
- Index ls output with index size and percent unmerged
- Update the Go client to a released version
^ I will do all of these before making a release.

### CODE REVIEW

File: cmd/flags/hnsw.go [hunk 1]
```diff
@@ -7,29 +7,37 @@ import (
...
+		MaxReindexRecords: Uint32OptionalFlag{},
+		ReindexInterval:   DurationOptionalFlag{},
 	}
 }
 
 func (cf *BatchingFlags) NewFlagSet() *pflag.FlagSet {
 	flagSet := &pflag.FlagSet{}
-	flagSet.Var(&cf.MaxRecords, BatchMaxRecords, "Maximum number of records to fit in a batch.")              //nolint:lll // For readability
-	flagSet.Var(&cf.Interval, BatchInterval, "The maximum amount of time to wait before finalizing a batch.") //nolint:lll // For readability
+	flagSet.Var(&cf.MaxIndexRecords, BatchMaxIndexRecords, "Maximum number of records to fit in a batch.")                           //nolint:lll // For readability
```
Comment (reugn) [inline cmd/flags/hnsw.go:27]: `//nolint:lll` can be set once, at the top of the function.

File: cmd/writers/indexList.go [hunk 2]
```diff
@@ -75,17 +103,20 @@ func (itw *IndexTableWriter) AppendIndexRow(
...
 				{"MaxMemQueueSize*", v.HnswParams.GetMaxMemQueueSize()},
-				{"Batch Max Records*", v.HnswParams.BatchingParams.GetMaxRecords()},
-				{"Batch Interval*", convertMillisecondToDuration(uint64(v.HnswParams.BatchingParams.GetInterval()))},
-				{"Cache Max Entires*", v.HnswParams.CachingParams.GetMaxEntries()},
-				{"Cache Expiry*", convertMillisecondToDuration(v.HnswParams.CachingParams.GetExpiry())},
+				{"Batch Max Index Records*", v.HnswParams.BatchingParams.GetMaxIndexRecords()},
+				{"Batch Index Interval*", convertMillisecondToDuration(uint64(v.HnswParams.BatchingParams.GetIndexInterval()))},
+				{"Batch Max Reindex Records*", v.HnswParams.BatchingParams.GetMaxReindexRecords()},
+				{"Batch Reindex Interval*", convertMillisecondToDuration(uint64(v.HnswParams.BatchingParams.GetReindexInterval()))},
+				{"Cache Max Entries*", v.HnswParams.IndexCachingParams.GetMaxEntries()},
```
Comment (reugn) [inline cmd/writers/indexList.go:110]: Requires the typo fix in the tests.

**Review by reugn — APPROVED** [review]

File: cmd/flags/hnsw.go (hunk 1 above)
Comment (dwelch-spike) [reply cmd/flags/hnsw.go:27]: Oh cool, I didn't know that was possible. Changed.

File: cmd/writers/indexList.go (hunk 2 above)
Comment (dwelch-spike) [reply cmd/writers/indexList.go:110]: Good catch, changed.

**Pull request merged**